plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java

```diff
@@ -383,6 +359,92 @@ public BlobPath basePath() {
         return basePath;
     }
 
+    @Override
+    public boolean isReloadable() {
+        return true;
+    }
+
+    @Override
+    public void reload(RepositoryMetadata newRepositoryMetadata) {
+        // Reload configs for S3Repository
+        super.reload(newRepositoryMetadata);
+        repositoryMetadata = newRepositoryMetadata;
+        readRepositoryMetadata(repositoryMetadata);
+
+        // Reload configs for S3RepositoryPlugin
+        final Map<String, S3ClientSettings> clientsSettings = S3ClientSettings.load(repositoryMetadata.settings(), pluginConfigPath);
+        service.refreshAndClearCache(clientsSettings);
+        s3AsyncService.refreshAndClearCache(clientsSettings);
+
+        // Reload configs for S3BlobStore
+        BlobStore blobStore = getBlobStore();
+        blobStore.reload(repositoryMetadata);
```

Comment:
In the S3Service class, lets check if we need to make the `clientsCache`, `derivedClientSettings` as concurrent maps.